SIPS/sip-29.md

```diff
@@ -145,6 +164,46 @@ type OnAssetsConversionResponse = {
 };
 ```
 
+### Get Assets historical price
+
+```typescript
+import { OnAssetHistoricalPriceHandler } from "@metamask/snaps-sdk";
+
+export const onAssetsHistoricalPrice: OnAssetsHistoricalPriceHandler = async ({
+  from, to
+}) => {
+  const historicalPrice = /* Get historical price for given `from` and `to` */;
+  return { historicalPrice };
+};
+```
+The type for an `onAssetHistoricalPrice` handler function’s arguments is:
+
+```typescript
+interface OnAssetHistoricalPriceArguments {
+  from: Caip19AssetType;
+  to: Caip19AssetType;
+}
+```
+
+The type for an `onAssetHistoricalPrice` handler function’s return value is:
+
+```typescript
+type OnAssetHistoricalPriceResponse = {
+  // The UNIX timestamp of when the historical price was last updated.
+  updateTime: number;
+  // The UNIX timestamp of when the historical price will expire.
+  expirationTime?: number;
+
+  intervals: {
+    // The `all` value is a special interval that represents all available data.  
+    all?: number;
```

Comment:
Wait, I don't understand this type, what is an example value of this?

Reply:
Sorry, copy paste things :sweat_smile: